CriteoPublisherSdk/Tests/UnitTests/CR_BidManagerTests.m

```diff
@@ -253,77 +266,388 @@ - (void)testGetBid_GivenLiveBiddingIsDisabled_ThenGetBidThenFetch {
+  [self fetchLiveBidForAdUnit:self.adUnitForEmptyBid
+             andExpectCdbCall:NO
+                    bidCached:nil
+                  bidConsumed:nil
+                 bidResponded:nil];
+
+  XCTAssertEqual(self.cacheManager.bidCache[self.adUnit1], silentBid);
+}
+
+- (void)testLiveBid_GivenExpiredSilentMode_ThenCdbCalled_AndResponseGiven {
+  // Expired user silent, 1s before now
+  self.bidManager.cdbTimeToNextCall =
+      [[NSDate dateWithTimeIntervalSinceNow:-1] timeIntervalSinceReferenceDate];
+
+  [self fetchLiveBidAndExpectBidCached:nil bidConsumed:nil bidResponded:self.bid1];
+}
+
+- (void)testLiveBid_GivenResponseErrorAndSilentModeAndValidBidInCache_ThenResponseGiven {
+  [self givenApiHandlerRespondError:NSError.new
+              doingBeforeResponding:^{
+                [self givenUserSilenced];
+              }];
+
+  [self fetchLiveBidAndExpectBidCached:nil bidConsumedAndResponded:self.bid1];
+}
+
+- (void)givenApiHandlerRespondWithUserSilence {
+  CR_CdbResponse *cdbResponseMock = OCMClassMock(CR_CdbResponse.class);
+  OCMStub(cdbResponseMock.timeToNextCall).andReturn(123);
+  [self givenApiHandlerRespond:cdbResponseMock
+         doingBeforeResponding:^{
+         }];
+}
+
+- (void)testLiveBid_GivenSilentUserResponse_ThenUserSilenceUpdated {
+  [self givenApiHandlerRespondWithUserSilence];
+
+  [self fetchLiveBidForAdUnit:self.adUnit1];
+
+  XCTAssertTrue(self.bidManager.isInSilenceMode);
+}
+
+- (void)testLiveBid_GivenSilentUserResponseAfterTimeBudget_ThenUserSilenceUpdated {
+  [self givenApiHandlerRespondWithUserSilence];
+  [self givenTimeBudgetExceeded];
+
+  [self fetchLiveBidForAdUnit:self.adUnit1];
+
+  XCTAssertTrue(self.bidManager.isInSilenceMode);
+}
+
+- (void)testLiveBid_GivenNotSilentUserResponse_ThenUserSilenceNotUpdated {
+  [self givenApiHandlerRespondBid:[self validCdbBidForAdUnit:self.adUnit1]
+            doingBeforeResponding:^{
+              [self givenUserSilenced];
+            }];
+
+  [self fetchLiveBidForAdUnit:self.adUnit1];
+
+  XCTAssertTrue(self.bidManager.isInSilenceMode);
+}
+
+- (void)testLiveBid_GivenNotSilentUserResponseAfterTimeBudget_ThenUserSilenceNotUpdated {
+  [self givenApiHandlerRespondBid:[self validCdbBidForAdUnit:self.adUnit1]
+            doingBeforeResponding:^{
+              [self givenUserSilenced];
+            }];
+
+  [self fetchLiveBidForAdUnit:self.adUnit1];
+
+  XCTAssertTrue(self.bidManager.isInSilenceMode);
+}
+
+#pragma mark Silent slot
+
+- (void)testLiveBid_GivenSilentBidInCache_ThenCdbNotCalledAndNoResponseGiven {
+  CR_CdbBid *silentBid = CR_CdbBidBuilder.new.adUnit(self.adUnit1).silenced().build;
+  self.bid1 = silentBid;
+  self.cacheManager.bidCache[self.adUnit1] = self.bid1;
+  [self givenApiHandlerRespondValidBid];
+
+  [self fetchLiveBidForAdUnit:self.adUnit1
+             andExpectCdbCall:NO
+                    bidCached:nil
+                  bidConsumed:nil
+                 bidResponded:nil];
 
   // Silent bid from cdb call has been cached
```

Comment:
The comment does not represent the expectation.
Isn't the expectation: silenced is not consumed from cache ?

Reply:
This is a leftover of the test before, will edit the comment. Also this is a bit redundant with the bidConsumed expect above.

Reply:
Done